File: java/src/main/java/deltix/luminary/implementation/FileNode.java
package deltix.luminary.implementation;

import deltix.luminary.Location;
import deltix.luminary.NameValuePair;

import java.util.ArrayList;
import java.util.List;

public class FileNode extends TypeContainerNode {
    private final List<ImportNode> imports = new ArrayList<>();
    private final List<NameValuePair> options = new ArrayList<>();

    private Location location;
    private String namespace;

    public FileNode() {}

    public Location getLocation() {
        return location;
    }

    public void setLocation(Location location) {
        this.location = location;
    }

    public String getNamespace() {
        return namespace;
    }

    public void setNamespace(String namespace) {
        this.namespace = namespace;
    }

    public List<ImportNode> getImports() {
        return imports;
    }

    public List<NameValuePair> getOptions() {
        return options;
    }
}

File: csharp/main/Implementation/FileNode.cs
using System;
using System.Collections.Generic;

namespace Deltix.Luminary.Implementation
{
	internal class FileNode : ITypeContainerNode
	{
		public String FilePath { get; set; }

		public String Namespace { get; set; }

		public List<ImportNode> Imports { get; } = new List<ImportNode>();

		public List<NameValuePair> Options { get; } = new List<NameValuePair>();

		public List<ClassNode> DefinedClasses { get; } = new List<ClassNode>();

		public List<EnumerationNode> DefinedEnumerations { get; } = new List<EnumerationNode>();

		public List<InterfaceNode> DefinedInterfaces { get; } = new List<InterfaceNode>();

		public List<DecoratorNode> DefinedDecorators { get; } = new List<DecoratorNode>();
	}
}
